zstash/extract.py

```diff
@@ -350,7 +350,7 @@ def multiprocess_extract(
     workers_to_matches: List[List[FilesRow]] = [[] for _ in range(num_workers)]
     for db_row in matches:
         tar = db_row.tar
-        workers_idx: int
+        worker_idx: int
```

Comment:
This rename was to match the index variable in the following for loop, but it was obscuring a different error (redundant variable definition) which `mypy` is picking up on now. I'll leave this for somebody who knows the code better to sort out, but it seems like a pretty easy fix.

Reply:
Ah ok so, the whole relevant block on the `main` branch` is
```
    # Using a greedy approach, populate workers_to_tars.
    for _, tar in enumerate(tar_to_size):
        # The worker with the least work should get the current largest amount of work.
        workers_work: int
        worker_idx: int
        workers_work, worker_idx = heapq.heappop(work_to_workers)
        workers_to_tars[worker_idx].add(tar)
        # Add this worker back to the heap, with the new amount of work.
        worker_tuple: Tuple[float, int] = (workers_work + tar_to_size[tar], worker_idx)
        # FIXME: error: Cannot infer type argument 1 of "heappush"
        heapq.heappush(work_to_workers, worker_tuple)  # type: ignore

    # For worker i, workers_to_matches[i] is a list of
    # matches from the database for it to process.
    workers_to_matches: List[List[FilesRow]] = [[] for _ in range(num_workers)]
    for db_row in matches:
        tar = db_row.tar
        workers_idx: int
        for worker_idx in range(len(workers_to_tars)):
            if tar in workers_to_tars[worker_idx]:
                # This worker gets this db_row.
                workers_to_matches[worker_idx].append(db_row)
```

So yeah once `workers_idx: int` is changed to `worker_idx: int` to actually match the variable in this loop, `mypy` complains because we're attempting to re-defined a variable from the previous loop. It looks like in practice that doesn't actually matter since each occurrence of `worker_idx` stays inside its scoping (i.e. its loop), but `mypy` doesn't have a way to know that.

I think here I'd say to actually keep it defined as `workers_idx: int` and change that one loop accordingly:
```
        workers_idx: int
        for workers_idx in range(len(workers_to_tars)):
            if tar in workers_to_tars[workers_idx]:
                # This worker gets this db_row.
                workers_to_matches[workers_idx].append(db_row)
```

Reply:
OK, I will do that!